### DISCUSSION TITLE
Dynamically Load Contract based on Network

### DISCUSSION THREAD

**Discussion by lsbyerley** [wevm/wagmi#244] [Q&A]
Is it possible to prevent a runtime error when passing in null as the `addressOrName` for the `useContract` hook? I would like to switch contract addresses when changing the network and a runtime error is shown if null is passed in.

```javascript
const contractAddressConfig = (chainId) => {
  const config = {
    4: {
      name: 'rinkeby',
      address: RINKEBY_CONTRACT,
    },
    5: {
      name: 'goerli',
      address: GOERLI_CONTRACT,
    },
  };
  return config[chainId] || null;
};

const [{ data: networkData }, switchNetwork] = useNetwork();
const chainId = networkData?.chain?.id;

// Passing in 4 on first load works fine, but switching networks to something other than 4 or 5 breaks
const contract = useContract({
  addressOrName: contractAddressConfig(chainId),
  contractInterface: JSON.abi,
  signerOrProvider: signerData,
});

// called via an onClick action
const onChangeNetwork = async (c) => {
  await switchNetwork(c.id);
};
```
When changing networks, there is a runtime error for the useContract hook:
`Error: invalid contract address or ENS name (argument="addressOrName", value=null, code=INVALID_ARGUMENT, version=contracts/5.6.0)`

**Comment by FreezyEx** [comment]
I am experiencing the same issue

**Comment by tmm** [comment]
This now works with `wagmi@0.7` and above.
